packages/cli/src/commands/worker.ts

```diff
@@ -360,9 +355,38 @@ export class Worker extends BaseCommand {
 				},
 			);
 
-			server.listen(port, () => {
-				this.logger.info(`\nn8n worker health check via, port ${port}`);
-			});
+			let presetCredentialsLoaded = false;
+			const endpointPresetCredentials = config.getEnv('credentials.overwrite.endpoint');
+			if (endpointPresetCredentials !== '') {
+				// POST endpoint to set preset credentials
+				app.post(
+					`/${endpointPresetCredentials}`,
+					async (req: express.Request, res: express.Response) => {
+						if (!presetCredentialsLoaded) {
+							const body = req.body as ICredentialsOverwrite;
+
+							if (req.contentType !== 'application/json') {
```

Comment:
I think we might need to add bodyparser here. definitely need to test this first.